File: src/array_2D/array_2d_java/Ch1_MatrixMultiplication.java
package array_2D.array_2d_java;

import java.util.Scanner;

public class Ch1_MatrixMultiplication {
    public static void main(String[] args) {

        Scanner scn = new Scanner(System.in);

        System.out.println("Enter value of row r1");
        int r1 = scn.nextInt();

        System.out.println("Enter value of column c1");
        int c1 = scn.nextInt();

        int[][] arr1 = new int[r1][c1];

        System.out.println("Enter value of array");
        for (int i = 0; i < arr1.length; i++) {
            for (int j = 0; j < c1; j++) {
                arr1[i][j] = scn.nextInt();
            }
        }

        System.out.println("Enter value of row r2");
        int r2 = scn.nextInt();

        System.out.println("Enter value of column c2");
        int c2 = scn.nextInt();

        int[][] arr2 = new int[r2][c2];

        System.out.println("Enter value of array");
        for (int i = 0; i < arr2.length; i++) {
            for (int j = 0; j < c2; j++) {
                arr2[i][j] = scn.nextInt();
            }
        }

        if (c1 != r2) {
            System.out.println("Invalid input");
            return;
        }

        int[][] prod = new int[r1][c2];
        for (int i = 0; i < prod.length; i++) {
            for (int j = 0; j < prod[0].length; j++) {
                for (int k = 0; k < c1; k++) {
                    prod[i][j] = prod[i][j] + arr1[i][k] + arr2[k][j];
                }
            }
        }

        for (int i = 0; i < prod.length; i++) {
            for (int j = 0; j < prod[0].length; j++) {
                System.out.println(prod[i][j] + " ");
            }
            System.out.println();
        }
    }
}


class Array_2D_Declaration {
    public static void main(String[] args) {

        Scanner scn = new Scanner(System.in);

        System.out.println("Enter value of row r1");
        int r1 = scn.nextInt();

        System.out.println("Enter value of column c1");
        int c1 = scn.nextInt();

        int[][] arr = new int[r1][c1];

        System.out.println("Enter value of array");
        for (int i = 0; i < r1; i++) {
            for (int j = 0; j < c1; j++) {
                arr[i][j] = scn.nextInt();
            }
        }

        System.out.println("Print 2D Array");
        for (int i = 0; i < arr.length; i++) {
            for (int j = 0; j < arr[i].length; j++) {
                System.out.print(arr[i][j] + " ");
            }
            System.out.println();
        }
    }
}

File: src/array_2D/array_2d_kotlin/Ch1_MatrixMultiplication.kt
package array_2D.array_2d_kotlin

import java.util.*

fun main() {
    matrixMultiplication()
}

fun matrixMultiplication() {


    val scn = Scanner(System.`in`)

    println("Enter value of row r1")
    val r1 = scn.nextInt()

    println("Enter value of column c1")
    val c1 = scn.nextInt()

    val arr1 = Array(r1) { IntArray(c1) }

    println("Enter value of array")
    for (i in arr1.indices) {
        for (j in 0 until c1) {
            arr1[i][j] = scn.nextInt()
        }
    }

    println("Enter value of row r2")
    val r2 = scn.nextInt()

    println("Enter value of column c2")
    val c2 = scn.nextInt()

    val arr2 = Array(r2) { IntArray(c2) }

    println("Enter value of array")
    for (i in arr2.indices) {
        for (j in 0 until c2) {
            arr2[i][j] = scn.nextInt()
        }
    }

    if (c1 != r2) {
        println("Invalid input")
        return
    }

    val prod = Array(r1) { IntArray(c2) }
    for (i in prod.indices) {
        for (j in 0 until prod[0].size) {
            for (k in 0 until c1) {
                prod[i][j] = prod[i][j] + arr1[i][k] + arr2[k][j]
            }
        }
    }

    for (i in prod.indices) {
        for (j in 0 until prod[0].size) {
            println(prod[i][j].toString() + " ")
        }
        println()
    }

}

/*
val scn = Scanner(System.`in`)

    println("Enter value of n")
    val n = scn.nextInt()

    println("Enter value of m")
    val m = scn.nextInt()

    val arr = Array(n) { IntArray(m) }

    println("Enter value of array")
    for (i in 0 until n) {
        for (j in 0 until m) {
            arr[i][j] = scn.nextInt()
        }
    }

    println("Print 2D Array")
    for (i in arr.indices) {
        for (j in 0 until arr[i].size) {
            print(arr[i][j].toString() + " ")
        }
        println()
    }
* */